nexus/reconfigurator/planning/src/mgs_updates/mod.rs

```diff
@@ -63,33 +79,44 @@ pub fn plan_mgs_updates(
             Ok(MgsUpdateStatus::Done) => {
                 info!(
                     log,
-                    "SP update completed \
-                     (will remove it and re-evaluate board)";
-                    update
-                );
-                boards_preferred.insert(update.baseboard_id.clone());
-            }
-            Ok(MgsUpdateStatus::Impossible) => {
-                info!(
-                    log,
-                    "SP update impossible \
+                    "MGS-driven update completed \
                      (will remove it and re-evaluate board)";
                     update
                 );
                 boards_preferred.insert(update.baseboard_id.clone());
             }
+            Ok(MgsUpdateStatus::Impossible) => match impossible_update_policy {
+                ImpossibleUpdatePolicy::Keep => {
+                    info!(
+                        log,
+                        "keeping apparently-impossible MGS update (waiting \
```

Comment:
How about changing all of the "MGS update" for "MGS-driven update"? Otherwise it sounds like we're updating MGS itself no?

```suggestion
                        "keeping apparently-impossible MGS-driven update (waiting \
```